src/server/game/Maps/Map.cpp

```diff
@@ -678,6 +678,20 @@ void Map::Update(const uint32 t_diff)
         player->Update(t_diff);
 
         VisitNearbyCellsOf(player, grid_object_update, world_object_update);
+
+        // Handle updates for creatures in combat with player and are more than 60 yards away
+        if (player->IsInCombat())
+        {
+            HostileReference* ref = player->getHostileRefManager().getFirst();
+
+            while (ref)
+            {
+                if (Unit* unit = ref->GetSource()->GetOwner())
+                    if (unit->GetTypeId() == TYPEID_UNIT && !unit->IsWithinDist2d(player->PositionXYStream().Pos, 60.0f))
```

Comment:
I am wondering whether there is a point to checking the distance. Since, if the cell has already been visited (by the player iterator above for example) then it won't be visited again anyway.

Maybe someone knows how expensive the distance check vs the checks in the Visit route are and can advise?